### PR TITLE
Elide lifetimes

### PR DESCRIPTION
**Pull request by JustusFluegel**
Fixes new clippy lints on nightly & beta.

### CODE REVIEW

File: packages/push/src/push_vm/stack.rs [hunk 1]
```diff
@@ -411,10 +411,10 @@ impl<T> Stack<T> {
     {
         let iter = iter.into_iter();
         // Check that adding these items won't overflow the stack.
-        if !iter
+        if iter
             .len()
             .checked_add(self.size())
-            .is_some_and(|x| x <= self.max_stack_size)
+            .is_none_or(|x| x > self.max_stack_size)
```
Comment (NicMcPhee) [inline packages/push/src/push_vm/stack.rs:417]: Nice – that's definitely much better!

**Review by NicMcPhee — APPROVED** [review]
Nice – thanks!

File: packages/push/src/push_vm/stack.rs (hunk 1 above)
Comment (JustusFluegel) [reply packages/push/src/push_vm/stack.rs:417]: Yeah that is what that does, they kind of depend on each other, #277 is based on top of this branch so we can either merge them chronologically or we can just merge #277  and then close this as well.

**Comment by NicMcPhee** [comment]
These changes were part of #277, right? So we can/should just close this one?

**Pull request closed without merging**